src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs

```diff
@@ -273,6 +273,10 @@ private static bool IsCodeActionNotSupportedByLSP(IUnifiedSuggestedAction sugges
                             applicableRange: nestedActionSet.ApplicableToSpan.HasValue
                                 ? ProtocolConversions.TextSpanToRange(nestedActionSet.ApplicableToSpan.Value, documentText) : null,
                             nestedSetNumber, codeActionPath, ref currentHighestSetNumber));
+
+                        // Once CodeActions at this nesting have been added, remove the last element from the code action path
+                        // so that the next nested action can be added to the correct path.
+                        codeActionPath.RemoveLast();
```

Comment:
i don't quite get it.  shouldn't it that at thee nd of this entire method, we pop off the last item we added?  

Also, sholud the code-action-path be immutable to ensure we're not mutating in place something poitned to by lots of items?